ml-agents/mlagents/trainers/buffer.py

```diff
@@ -217,6 +220,48 @@ def make_mini_batch(self, start, end):
                 mini_batch[key] = np.array(self[key][start:end])
             return mini_batch
 
+        # SAC HAC
+        def sample_mini_batch(self, batch_size):
+            """
+            Creates a mini-batch from a random start and end.
+            : param batch_size: number of elements to withdraw.
+            """
+            mini_batch_lists = {}
+            mini_batch = Buffer.AgentBuffer()
+            idxes = [
```

Comment:
So it's OK if the same index occurs multiple times, right?

Reply:
Yes, I think generally experience replay buffers sample with replacement. It's unlikely to change the training performance, though, as there's usually 10,000's of things in the buffer and batch size in the 100's.

If we're enforcing Python > 3.6 (we are, right?) we can use `random.choice()` as well.